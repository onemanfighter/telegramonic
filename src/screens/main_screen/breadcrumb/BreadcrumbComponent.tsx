import { Box, Breadcrumb, Button, Text } from '@chakra-ui/react';
import { useTranslation } from 'react-i18next';
import { Link, useLocation } from 'react-router-dom';

const BreadcrumbComponent = () => {
  const location = useLocation();
  const { t } = useTranslation();

  const onBreadcrumbClick = () => {
    console.log('Breadcrumb clicked');
  };

  const paths = location.pathname.split('/').filter((path) => path !== '');
  return (
    <Box
      display={'flex'}
      paddingX={'12'}
      paddingY={'2'}
      bgColor={'grey.500'}
      border={'1px solid grey'}
      borderRadius={'lg'}
      marginX={4}
      marginTop={2}
      flexDirection={'row'}
      justifyContent={'start'}
    >
      <Button
        as={Link}
        to={`/home`}
        onClick={onBreadcrumbClick}
        size={'sm'}
        variant={'outline'}
        colorScheme="blue"
        marginX={1}
      >
        {t('home')}
      </Button>
      <Text
        fontSize={'large'}
        fontWeight={'400'}
        textAlign={'center'}
        marginX={1}
      >
        {' / '}
      </Text>
      <Breadcrumb separator={' / '}>
        {paths.map((path, index) => (
          <Button
            key={index}
            onClick={onBreadcrumbClick}
            size={'sm'}
            as={Link}
            to={`/${path}`}
            variant={'outline'}
            colorScheme="blue"
            marginX={1}
          >
            {path}
          </Button>
        ))}
      </Breadcrumb>
    </Box>
  );
};

export default BreadcrumbComponent;
