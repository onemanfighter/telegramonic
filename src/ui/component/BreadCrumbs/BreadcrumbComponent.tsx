import Breadcrumb from 'react-bootstrap/Breadcrumb';
import styles from './BreadcrumbComponent.module.css'

function BreadcrumbComponent() {
  return (
    <Breadcrumb className={styles.breadcrumb}>
      <Breadcrumb.Item className={styles.breadcrumbItem} href="#" >Home</Breadcrumb.Item>
      <Breadcrumb.Item className={styles.breadcrumbItem} href="https://getbootstrap.com/docs/4.0/components/breadcrumb/">
        Library
      </Breadcrumb.Item>
      <Breadcrumb.Item className={styles.breadcrumbItem}  active>Data</Breadcrumb.Item>
    </Breadcrumb>
  );
}

export default BreadcrumbComponent;