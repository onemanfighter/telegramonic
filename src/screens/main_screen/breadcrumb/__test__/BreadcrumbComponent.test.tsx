import { render } from '@testing-library/react';
import BreadcrumbComponent from '../BreadcrumbComponent';

jest.mock('react-router-dom', () => ({
  useLocation: () => ({
    pathname: '/home/first/second/third',
  }),
}));

describe('BreadcrumbComponent', () => {
  it('should render correctly', () => {
    const { container } = render(<BreadcrumbComponent />);

    expect(container).toMatchSnapshot();
  });
});
