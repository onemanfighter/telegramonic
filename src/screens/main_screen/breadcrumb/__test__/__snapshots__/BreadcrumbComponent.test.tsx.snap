// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbComponent should render correctly 1`] = `
<div>
  <div
    class="css-qc0wyl"
  >
    <button
      class="chakra-button css-1qr028o"
      to="/home"
      type="button"
    >
      Home
    </button>
    <p
      class="chakra-text css-tfsct8"
    >
       / 
    </p>
    <nav
      aria-label="breadcrumb"
      class="chakra-breadcrumb css-0"
    >
      <ol
        class="chakra-breadcrumb__list css-70qvj9"
      >
        <button
          class="chakra-button css-1qr028o"
          separator=" / "
          spacing="0.5rem"
          to="/home"
          type="button"
        >
          home
        </button>
        <button
          class="chakra-button css-1qr028o"
          separator=" / "
          spacing="0.5rem"
          to="/first"
          type="button"
        >
          first
        </button>
        <button
          class="chakra-button css-1qr028o"
          separator=" / "
          spacing="0.5rem"
          to="/second"
          type="button"
        >
          second
        </button>
        <button
          class="chakra-button css-1qr028o"
          separator=" / "
          spacing="0.5rem"
          to="/third"
          type="button"
        >
          third
        </button>
      </ol>
    </nav>
  </div>
</div>
`;
